src/Microsoft.Identity.Web/DownstreamWebApiSupport/IDownstreamWebApi.cs

```diff
@@ -28,19 +29,22 @@ public interface IDownstreamWebApi
         /// will find the user from the HttpContext.</param>
         /// <param name="content">HTTP context in the case where <see cref="DownstreamWebApiOptions.HttpMethod"/> is
         /// <see cref="HttpMethod.Patch"/>, <see cref="HttpMethod.Post"/>, <see cref="HttpMethod.Put"/>.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
         /// <returns>An <see cref="HttpResponseMessage"/> that the application will process.</returns>
         public Task<HttpResponseMessage> CallWebApiForUserAsync(
             string serviceName,
             Action<DownstreamWebApiOptions>? calledDownstreamWebApiOptionsOverride = null,
             ClaimsPrincipal? user = null,
-            StringContent? content = null)
+            StringContent? content = null,
+            CancellationToken cancellationToken = default)
```

Comment:
Is this not a breaking change? If I have an implementation for IDownstreamApi (e.g.for test), it will be broken by this change?

Reply:
Ah, you're right. I only thought of consumers using the interface through the `AddDownstreamWebApi`, not of implementors. I guess it will have to wait for the next major?

Reply:
I'll let the package owners (@jmprieur  and @jennyf19 ) answer that, but ideally we should not make interface changes. Alternatives are: 

- extension methods
- default interface methods https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods

Reply:
I tried having a look at this;
**Extension methods**
I don't think this would work, as this needs to happen through the implementation to be of use.

**`ICancellableDownstreamWebApi : IDownstreamWebApi`**
Could technically work, but it would have to be registered twice with DI, and would be too clunky as a consumer imo.

**Default Interface methods**
I thought this would work. I tried having default interface methods for the cancellationToken versions of the methods point to the ones without, and do it in reverse in the implementation. That way it would be opt-in for implementors to accept cancellationTokens, and the default implementation would accept it.
The problem was that having these two definitions would make the many existing calls ambiguous
```csharp
// Simplified interface
Task CallForUserAsync(string url, string? scheme = null);
Task CallForUserAsync(string url, string? scheme = null, CancellationToken cancellationToken = default);

// This call would be ambigous
api.CallForUserAsync("/endpoint");
```
I could probably fix it by having the CancellationToken not be an optional param, but then I would have to order it before the optional params. I don't know if it's an actual rule, but everywhere I've seen a cancellationToken, it's appeared last, so I assume it's preferred to have it as a last param.
Either way, there would be some code complexity added, with default interface methods calling each other, and the implementation the other way.

**TL:DR**
To not break for `IDownstreamWebApi` implementors, there would either have to be some code complexity added to this project, or it would be clunky for consumers to use (having two interfaces to choose from). IMO it's not worth the effort needed, to not break for implementors, even though it would be desired.

I'll leave it up to the project owners how they'll want to do it, if they want this feature.